staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/celcoststability_test.go

```diff
@@ -359,12 +359,9 @@ func TestCelCostStability(t *testing.T) {
 				"!has(self.val.c)":                               3,
 				"has(self.val.d)":                                2,
 				"self.val.all(k, self.val[k] > 0)":               17,
-				"self.val.exists(k, self.val[k] > 1)":            19,
 				"self.val.exists_one(k, self.val[k] == 2)":       14,
-				"!self.val.exists(k, self.val[k] > 2)":           20,
 				"!self.val.exists_one(k, self.val[k] > 0)":       17,
 				"size(self.val) == 2":                            4,
-				"self.val.map(k, self.val[k]).exists(v, v == 1)": 43,
```

Comment:
is the cost non-deterministic because it short-circuits as soon as it hits the 1?

Reply:
Correct, this is a short-circuiting operation.  exists_one must do a full iteration

Reply:
all() can also short circuit if it fails on any elements, but the only case with it doesn't do that so it's consistent.